.github/workflows/pr-checks.yml

```diff
@@ -58,6 +62,9 @@ jobs:
         uses: codecov/codecov-action@v3
 
   docker-local:
+    permissions:
+      contents: read  # for docker/build-push-action to read repo content
```

Comment:
```suggestion
```

It's granted at the top level.

Reply:
Committing the suggestion because I have never bothered trying to confirm my own understanding.

AFAIK the permissions are set in a single-go, not with inheritance. I think my suggestion is necessary, but I'm quite happy to try the alternative to find out whether I've been doing this redundantly.